### PR TITLE
gRPC: bump version, update dependencies

### PR DESCRIPTION
**Pull request by Jihadist**
Specify library name and version:  **grpc/1.38.0**

Updating dependency protobuf due to https://github.com/protocolbuffers/protobuf/commit/5d0c30934bb6a6c1d387b549f5af3a4efcebe2ef
After this commit you can use ```protobuf_generate``` cmake function with grpc_cpp_plugin 
Updating dependency re2: i have no reason, but i update one dependency so i update another too.

Update library version too.
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/grpc/all/conanfile.py [hunk 1]
```diff
@@ -236,6 +236,7 @@ def package_info(self):
         # gRPC::grpc_cpp_plugin
         if self.options.cpp_plugin:
             module_target_rel_path = os.path.join("lib", "cmake", "grpc_cpp_plugin.cmake")
+            self.cpp_info.components["execs"].builddirs = os.path.join("lib", "cmake")
```
Comment (SpaceIm) [inline recipes/grpc/all/conanfile.py:239]:
```suggestion
            self.cpp_info.components["execs"].builddirs.append(os.path.join("lib", "cmake"))
```

File: recipes/grpc/all/conanfile.py [hunk 2]
```diff
@@ -55,10 +55,10 @@ def _build_subfolder(self):
     def requirements(self):
         self.requires('zlib/1.2.11')
         self.requires('openssl/1.1.1k')
-        self.requires('protobuf/3.15.5')
+        self.requires('protobuf/3.17.1')
         self.requires('c-ares/1.17.1')
         self.requires('abseil/20210324.0')
```
Comment (SpaceIm) [inline recipes/grpc/all/conanfile.py:60]:
```suggestion
        self.requires('abseil/20210324.1')
```

**Comment by Jihadist** [comment]
I tried locally to reproduce compilation error (a4171e40bfab329284d0dde545448b40f2d88315) with this config:
```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MT
compiler.version=16
os=Windows
os_build=Windows
```
But i dont receive any errors.

**Comment by SpaceIm** [comment]
close/open PR, it should work

**Review by Croydon — APPROVED** [review]

File: recipes/grpc/all/conandata.yml [hunk 3]
```diff
@@ -1,4 +1,7 @@
 sources:
+  "1.38.0":
+    url: "https://github.com/grpc/grpc/archive/refs/tags/v1.38.0.tar.gz"
+    sha256: "ABD9E52C69000F2C051761CFA1F12D52D8B7647B6C66828A91D462E796F2AEDE"
```
Comment (SpaceIm) [inline recipes/grpc/all/conandata.yml:4]: cosmetic: keep it lowercase to make recipe consistent

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/grpc/all/conandata.yml (hunk 3 above)
Comment (Jihadist) [reply recipes/grpc/all/conandata.yml]: ok

**Review by prince-chrismc — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**